### PR TITLE
Only use KEYBOARD.COK when it is available, otherwise use embedded table

### LINKED ISSUE #315: Dethrace won't start on my VMWare/Fedora 38
its inside a VMWare vm - so no native opengl drivers available (i think)

build with Fedora 38/gcc 13.1.1 without any warning

just gives me that output

```
[linux@fedora __build]$ DETHRACE_ROOT_DIR=../Carma/ ./dethrace 
Dethrace version: v0.6.0-21-g34a22f2
Data directory: ../Carma/
Game mode: Carmageddon
warning: queue 0x22bac10 destroyed while proxies still attached:
  wl_display@1 still attached
[INFO] GLRenderer_Init OpenGL vendor string: Mesa
[INFO] GLRenderer_Init OpenGL renderer string: llvmpipe (LLVM 16.0.1, 256 bits)
[INFO] GLRenderer_Init OpenGL version string: 4.5 (Core Profile) Mesa 23.0.3
[INFO] GLRenderer_Init OpenGL shading language version string: 4.50
warning: queue 0x237a820 destroyed while proxies still attached:
  wl_callback@40 still attached
  wl_surface@38 still attached
warning: queue 0x22ba910 destroyed while proxies still attached:
  wl_display@1 still attached
[linux@fedora __build]$ 
```

SUSE/Tumbleweed

```
linux@localhost:~/tests/dethreace_dev/__build> DETHRACE_ROOT_DIR=../Carma/ ./dethrace 
Dethrace version: v0.6.0-21-g34a22f2
Data directory: ../Carma/
Game mode: Carmageddon
[INFO] GLRenderer_Init OpenGL vendor string: VMware, Inc.
[INFO] GLRenderer_Init OpenGL renderer string: SVGA3D; build: RELEASE;  LLVM;
[INFO] GLRenderer_Init OpenGL version string: 4.3 (Compatibility Profile) Mesa 23.0.2
[INFO] GLRenderer_Init OpenGL shading language version string: 4.30
linux@localhost:~/tests/dethreace_dev/__build> 
```

### PR DESCRIPTION
**Pull request by madebr**
- The Carmagedddon keyboard table was taken from `KEYBOARD.TOK` from the steam/gog release
- All tables are now commented alongside their meaning `KEY_xxx`
- Replace a few `KEY_SHIFT_LSHIFT` with `KEY_SHIFT_ANY`
  Because of this, #321 is triggered with both shift keys.

Fixes #315

### CODE REVIEW

File: src/harness/harness.c [hunk 1]
```diff
@@ -36,49 +36,884 @@ tHarness_game_config harness_game_config;
...
+    's',   /* KEY_S */
+    't',   /* KEY_T */
+    'u',   /* KEY_U */
+    'v',   /* KEY_V */
+    'w',   /* KEY_W */
+    'x',   /* KEY_X */
+    'z',   /* KEY_Y */
+    'y',   /* KEY_Z */
+    '^',   /* KEY_GRAVE */
+    -0x21, /* KEY_MINUS */
```
Comment (dethrace-labs) [inline src/harness/harness.c:309]: Is it supposed to be `-0x21`?

File: src/harness/harness.c [hunk 2]
```diff
@@ -36,49 +36,884 @@ tHarness_game_config harness_game_config;
 // Platform hooks
 tHarness_platform gHarness_platform;
 
-/* clang-format off */
+// Carmgeddon ASCII codes
```
Comment (dethrace-labs) [inline src/harness/harness.c:39]: how about putting these tables in a separate file like `ascii_tables.h` to avoid adding all this data to the code file?

Comment (dethrace-labs) [reply src/harness/harness.c:39]: I did something similar with `sdl2_scancode_to_dinput.h`

File: src/harness/harness.c (hunk 1 above)
Comment (madebr) [reply src/harness/harness.c:309]: I think this is a German Eszett (ẞ).
```python
b'\xdf'.decode("windows-1252")
```
The binary really contains `FFFFFFDFh`.
I think their compiler treated characters as signed ==> the 8-bit `0xdf` value is negative and must be sign-extended to a 32-bit integer.

File: src/harness/harness.c (hunk 2 above)
Comment (madebr) [reply src/harness/harness.c]: I created a `ascii_tables.h` header.

File: src/harness/harness.c (hunk 1 above)
Comment (madebr) [reply src/harness/harness.c]: The German DOS version contains the same codes, but they are all positive there.
So it looks like this is an oversight/bug of OG.

The text input behavior of German Dethrace and the German DOS version is now the same.
I didn't verify the German Windows executable.

File: src/DETHRACE/common/input.c [hunk 3]
```diff
@@ -403,13 +403,15 @@ int AddRollingLetter(char pChar, int pX, int pY, tRolling_type rolling_type) {
     let->current_offset = (gCurrent_graf_data->save_slot_letter_height * let->number_of_letters);
     for (i = 0; i < let->number_of_letters; i++) {
         if (rolling_type == eRT_numeric) {
-            let->letters[i] = pChar;
+            /* The (tU8) cast makes sure extended ASCII is positive. */
+            let->letters[i] = (tU8)pChar;
```
Comment (madebr) [inline src/DETHRACE/common/input.c:407]: This is done in the German Carmageddon DOS executable.

This makes sure the extended ASCII codes remain positive.
Before, these became negative here.

Not doing this causes an out-of-bounds access when reading the texture of an extended ASCII character.

File: src/harness/ascii_tables.h [hunk 4]
```diff
@@ -0,0 +1,893 @@
+#ifndef DETHRACE_ASCII_TABLES_H
```
Comment (madebr) [inline src/harness/ascii_tables.h:1]: This file was generated with the attached python script + json files: [ascii-codegen.zip](https://github.com/dethrace-labs/dethrace/files/11392696/ascii-codegen.zip)

**Comment by madebr** [comment]
If this pr is accepted, the new error message that is shown when no game data can be found is:
image

**Review by dethrace-labs — APPROVED** [review]
Nice change!

File: src/harness/ascii_tables.h [hunk 5]
```diff
@@ -0,0 +1,893 @@
...
+        's',        /* KEY_S */
+        't',        /* KEY_T */
+        'u',        /* KEY_U */
+        'v',        /* KEY_V */
+        'w',        /* KEY_W */
+        'x',        /* KEY_X */
+        'z',        /* KEY_Y */
+        'y',        /* KEY_Z */
+        '^',        /* KEY_GRAVE */
+        0xdf,       /* KEY_MINUS (ß) */
```
Comment (madebr) [inline src/harness/ascii_tables.h:280]: I didn't verify the actual textures. So there might be cases where the character in the comments does not match the actual glyph present in the texture.

**Pull request merged**